joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/TypeMap.scala

```diff
@@ -92,16 +94,40 @@ class TypeMap(astGenResult: AstGenRunnerResult, initialMappings: List[NamespaceT
     * types.
     */
   private def parseCompilationUnit(cu: DotNetNodeInfo): Map[String, Set[CSharpType]] = {
+    val namespaceTypeMap = mutable.Map.empty[String, Set[CSharpType]]
     cu.json(ParserKeys.Members)
       .arr
       .map(AstCreatorHelper.createDotNetNodeInfo(_))
       .filter { x =>
         x.node match
-          case NamespaceDeclaration => true
-          case _                    => false
+          case NamespaceDeclaration | FileScopedNamespaceDeclaration | ClassDeclaration => true
+          case _                                                                        => false
+      }
+      .foreach { parserNode =>
+        parserNode.node match
+          case NamespaceDeclaration | FileScopedNamespaceDeclaration => {
+            val (namespace, typesInNamespace) = parseNamespace(parserNode)
+            if (namespaceTypeMap.contains(namespace)) {
```

Comment:
Instead of this if-branch, we can use `updateWith`, e.g.
```scala
namespaceTypeMap.updateWith(namespace, {
   case Some(existing) => Option(existing ++ typesInNamespace)
   case None => Option(typesInNamespace)
})
```
And do a similar thing below

Reply:
Yup, that's the method I could not find. Thanks.